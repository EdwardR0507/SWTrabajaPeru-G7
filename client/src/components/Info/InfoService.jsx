import { React, useState } from "react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import { Container, Button, Typography } from "@material-ui/core/";
import DeleteIcon from "@material-ui/icons/Delete";
import ServiceModal from "../Modals/ServiceModal";

const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    marginTop: "20px",
    marginBottom: "20px",
    width: "80%",
    height: "25vh",
    borderBottom: "1px solid #00000012",
  },
  description: {
    display: "flex",
    flexDirection: "column",
    width: "100%",
    textAlign: "justify",
  },
  button: {
    width: "9em",
    height: "2.8em",
    fontSize: "0.9em",
    lineHeight: "16px",
    letterSpacing: "1.25px",
  },
  image: {
    maxWidth: "100%",
    objectFit: "cover",
  },
}));

const StyledContainerData = withStyles({
  root: {
    width: "100%",
    display: "flex",
    alignItems: "center",
  },
})(Container);

const StyledContainerImage = withStyles({
  root: {
    width: "25%",
  },
})(Container);

const StyledContainerButtons = withStyles({
  root: {
    display: "flex",
    width: "70%",
    justifyContent: "space-evenly",
    alignItems: "center",
  },
})(Container);

const WrapContainer = withStyles({
  root: {
    width: "100%",
    padding: "0.1em",
    overflowWrap: "break-word",
  },
})(Container);

const InfoService = ({ cat_nombre, ser_descripcion }) => {
  const classes = useStyles();

  const [descriptionService, setDescriptionService] = useState(ser_descripcion);
  const [state, setState] = useState(true);

  const [modalDescription, setModalDescription] = useState(ser_descripcion);

  const handleDelete = () => {
    setState(false);
  };

  const handleEdit = (e) => {
    setModalDescription(e.target.value);
  };

  return (
    <>
      {state ? (
        <Container className={classes.root}>
          <StyledContainerData>
            {/*Aquí irá la imagen del servicio, primero importamos la imagen y luego la colocamos dentro del src, no olvidar poner el alt */}
            <StyledContainerImage>
              <></>
              {/*<img src={imageService} alt={"servicio"} className={classes.image} />*/}
            </StyledContainerImage>
            <Container className={classes.description}>
              <Typography color="primary" variant="subtitle1">
                {cat_nombre}
              </Typography>
              <WrapContainer>
                <Typography variant="body2">{descriptionService}</Typography>
              </WrapContainer>
            </Container>
          </StyledContainerData>
          <StyledContainerButtons>
            <ServiceModal
              serviceDescription={descriptionService}
              handleEdit={handleEdit}
              modalDescription={modalDescription}
              setDescriptionService={setDescriptionService}
            />
            <Button
              variant="contained"
              endIcon={<DeleteIcon />}
              className={classes.button}
              onClick={handleDelete}
            >
              Eliminar
            </Button>
          </StyledContainerButtons>
        </Container>
      ) : null}
    </>
  );
};

export default InfoService;
