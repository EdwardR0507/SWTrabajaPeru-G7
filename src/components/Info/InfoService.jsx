import { React, useState } from "react";
import {
  makeStyles,
  withStyles,
  Container,
  Typography,
} from "@material-ui/core/";
import ServiceModal from "../Modals/ServiceModal";
import DialogDelete from "../Dialog/DialogDelete";
import { useLocation } from "react-router";
import { fetchData } from "../../services/services";
import image from "../../assets/services.jpg";
const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    marginTop: "20px",
    marginBottom: "20px",
    width: "80%",
    height: "25vh",
    borderBottom: "1px solid #00000012",
  },
  description: {
    display: "flex",
    flexDirection: "column",
    width: "100%",
    textAlign: "justify",
  },
  image: {
    maxWidth: "100%",
    objectFit: "cover",
  },
}));

const StyledContainerData = withStyles({
  root: {
    width: "100%",
    display: "flex",
    alignItems: "center",
  },
})(Container);

const StyledContainerImage = withStyles({
  root: {
    width: "25%",
  },
})(Container);

const StyledContainerButtons = withStyles({
  root: {
    display: "flex",
    width: "70%",
    justifyContent: "space-evenly",
    alignItems: "center",
  },
})(Container);

const WrapContainer = withStyles({
  root: {
    width: "100%",
    padding: "0.1em",
    marginBottom: "2em",
    overflowWrap: "break-word",
    display: "inline-block",
    wordBreak: "break-word"
  },
})(Container);

const InfoService = ({ cat_id, cat_nombre, ser_descripcion }) => {
  const classes = useStyles();

  const [descriptionService, setDescriptionService] = useState(ser_descripcion);
  //Controlar el renderizado condicional, al ser false no se renderizará el componente InfoService
  const [stateRender, setStateRender] = useState(true);

  const [modalDescription, setModalDescription] = useState(ser_descripcion);

  const location = useLocation();
  const state = location.state;

  const handleEdit = (e) => {
    setModalDescription(e.target.value);
  };

  const handleDelete = () => {
    const newData = { cat_id: cat_id };
    fetchData(state?.token, "POST", "service-auth", "DELETE_SERVICE", newData);
    setStateRender(false);
  };

  return (
    <>
      {stateRender ? (
        <Container className={classes.root}>
          <StyledContainerData>
            {/*Aquí irá la imagen del servicio, primero importamos la imagen y luego la colocamos dentro del src, no olvidar poner el alt */}
            <StyledContainerImage>
              <img src={image} alt={"servicio"} className={classes.image} />
            </StyledContainerImage>
            <Container className={classes.description}>
              <Typography color="primary" variant="subtitle1">
                {cat_nombre}
              </Typography>
              <WrapContainer>
                <Typography variant="body2">{descriptionService}</Typography>
              </WrapContainer>
            </Container>
          </StyledContainerData>
          <StyledContainerButtons>
            <ServiceModal
              cat_nombre={cat_nombre}
              serviceDescription={descriptionService}
              role="edit"
              handleEdit={handleEdit}
              modalDescription={modalDescription}
              setDescriptionService={setDescriptionService}
            />
            <DialogDelete
              role="delete"
              cat_nombre={cat_nombre}
              handleDelete={handleDelete}
            />
          </StyledContainerButtons>
        </Container>
      ) : null}
    </>
  );
};

export default InfoService;
