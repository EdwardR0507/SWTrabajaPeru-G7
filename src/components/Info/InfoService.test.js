/*Importamos las librerias principales*/
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import { Router } from 'react-router-dom';
import { createMemoryHistory  } from "history";
import InfoService from "./InfoService";
/*Renderizado del componente InfoService*/
describe("<InfoService />", () => {
  const history = createMemoryHistory()
  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
      push: mockHistoryPush, 
    }),
  }));
  const name = "nombre",description = "descripcion";
  let component;
  beforeEach(() => {
    component = render(<Router history={history}>
      <InfoService name={name} description={description} />
      </Router>);
  });
  test("renders InfoService", () => {
    component.getByText(name);
  });
  test("test handleEdit", () => {
    component.getByText(descripcion)
  })
  test("handleEdit test", () => {
    const wrapper = shallow(<Router history={history}>
      <InfoService />
    </Router>);
        const button = wrapper.find({ role: "edit" });
        button.simulate("click");
        expect(wrapper.state().modalDescription).toBe(false)
  })
  test("handleEdit test", () => {
    const wrapper = shallow(<Router history={history}>
      <InfoService />
    </Router>);
        const button = wrapper.find({ role: "delete" });
        button.simulate("click");
        expect(wrapper.state().stateRender).toBe(false)
  })
});
