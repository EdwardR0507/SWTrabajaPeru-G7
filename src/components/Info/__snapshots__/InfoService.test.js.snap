// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoService render "InfoService" 1`] = `ShallowWrapper {}`;
